src/cli/cli-report.c

```diff
@@ -583,19 +583,16 @@ static int run_event_on_dir_name_batch(
                 const char *dump_dir_name,
                 const char *event_name)
 {
-    int retval = -1;
-    problem_data_t *problem_data = NULL;
+    g_autoptr(problem_data_t) problem_data = NULL;
```

Comment:
While we're at it, couldn't we get rid of the `typedef` in `problem_data.h` and just use `GHashTable` instead of `problem_data_t`? But I don't have feel very strongly about it.

Reply:
Hm, I didn't notice that. However, although it's a `typedef`, there are multiple non-trivial functions around it (unlike the types we had eliminated before that were strictly just aliases), so I would keep it for now.